### ISSUE TITLE
I don't know if this can be concidered an issue.

### ISSUE THREAD

**Issue by DannN7** [dsoft20/psx_retroshader#4]
Hi, so I'm not familiar with shader programming, and I'm having some problems trying to Implement shadows on this shader, so is there a way you could do a version with the implementation of shadows? I'll be very glad if so, this would be a very good option for some types of game with old rendering style while still having shadows avaible.
As well Emission would be awesome.
Also if possible of giving some advice for flashlight (spot light) while using this shader, because it feels like the textures are ignoring the lights.
Thank you.
 -Dan.

**Comment by oxysoft** [comment]
can you link a PS1 game that had dynamic shadows?

**Comment by DannN7** [comment]
I think [Tekken 3](https://youtu.be/riUIbjZIsxQ?t=186) and [X-Men Mutant Academy](https://youtu.be/fuHLMoijqT0?t=110) had something like dynamic shadows (videos are linked), if not they are very impressive fake shadowing, I sadly don't know any other close to that, maybe tomb raider 5 (someone said on a forum that tomb raider had something close with flares on or while shooting)

Well, even if they had not, I could still make some use mixing old ps1 retro with shadows.
Edit: [Tony Hawk 2](https://youtu.be/LAbcCRYZhGw?t=172) as well seems to have it.

**Comment by oxysoft** [comment]
shadows like these weren't done using any of the fancy modern methods for rendering shadows but were actually done by projecting the model onto a texture and bolting it onto the stage, below the character, using just a quad. [Crash Bandicoot 3](https://youtu.be/HG-NRnGp3RA?t=1740) (and probably all of Naughty Dog's PS1 games) did the same thing as well. This isn't really something to add to the shaders in this repository and it shouldn't be too hard to do I think. I think you could do it with a RenderTexture, all you gotta do is have a camera with some image effects on it which turns the image black, and then move it around beneath your objcets every frame to capture some shadows for them.

**Comment by DannN7** [comment]
Yeah.. that could be archived for characters using RenderTexture and a camera and layers to ignore everything but the character itself.. Still what I wanted to archive was shadows for objects when using a spot or point light.. I guess I'll try to keep messing around with shaders until I'm able to get the results I need. 

Still, what about emission?
I could archive emission with another fake plane just above the area that I need emission but, some games like Alien Ressurection had what looks like emission map.
